Remove some legacy activity cruft from Aircraft

reaperrr opened the pull request:
You only need to look at the RTB activities to tell that `Aircraft` doing its own stuff here was somewhat redundant and just made things worse regarding debugging and code consistency.

Fixed some inconsistencies with the 'targetActor not available' case as well.

This is just a bit of clean-up before the upcoming larger refactor(s), so it would be nice to get this merged quickly.

reaperrr committed:
Remove some legacy activity cruft from Aircraft

You only need to look at the RTB activities to tell that
Aircraft doing its own stuff here was somewhat redundant
and just made things worse regarding debugging and
code consistency.

tovl commented:
Maybe this could be simplified further, considering that ReturnToBase and HeliReturnToBase already check if the targetActor is reserved?

reaperrr committed:
Make HeliReturnToBase use RTB.ChooseResupplier

Small consistency fix and prep for merging HRTB into RTB.

reaperrr commented:
You're right, the RTB activities should be able to handle that case on their own. Updated.

I also moved the SetLine to the activities, *after* the ChooseResupplier stuff, because that way the line is only set when the order actually succeeds and it simplifies the `Aircraft` code even further.

Furthermore, added a commit that makes HeliRTB use the `ChooseResupplier` from RTB.

tovl reviewed (dismissed):
Alright, looks good. Greatly improves DRYness. Also checked in game and as far as I can see this introduces no regressions. I don't know if my opinion counts for anything, but I think this can be merged.

pchote commented on OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs:
```diff
@@ -147,6 +147,8 @@ public override Activity Tick(Actor self)
 				}
 			}
 
+			self.SetTargetLine(Target.FromActor(dest), Color.Green);
```
Our conventions are that `SetTargetLine()` with `display: true` must *only* be called from `ResolveOrder`. This prevents improper visual feedback popping up when activities are queued from other activities.

This could stay, but needs to use the three argument version, and we will need to keep a call in `Aircraft` that gives the visual feedback.  The logic here will be simpler once #2598 is resolved.

reaperrr replied:
Reverted the target line changes as discussed on IRC.

pchote commented:
This is an internal refactoring with (hopefully!) not player or modder visible changes, so no changelog entry.
